server/src/main/java/org/opensearch/common/blobstore/transfer/RemoteTransferContainer.java

```diff
@@ -226,27 +233,7 @@ private long getActualChecksum() {
 
     @Override
     public void close() throws IOException {
-        if (inputStreams.get() == null) {
-            log.warn("Input streams cannot be closed since they are not yet set for multi stream upload");
-            return;
-        }
-
-        boolean closeStreamException = false;
-        for (InputStream is : Objects.requireNonNull(inputStreams.get())) {
-            try {
-                if (is != null) {
-                    is.close();
-                }
-            } catch (IOException ex) {
-                closeStreamException = true;
-                // Attempting to close all streams first before throwing exception.
-                log.error("Multipart stream failed to close ", ex);
-            }
-        }
-
-        if (closeStreamException) {
-            throw new IOException("Closure of some of the multi-part streams failed.");
-        }
+        closed.set(true);
```

Comment:
A couple things seem odd here:

1. This doesn't actually close the streams, but will just ensure subsequent access to the streams will throw AlreadyClosedExceptions (but still won't actually close them unless I'm misreading the something).
2. This `close()` method will do nothing if the `offsetRangeInputStream instanceof RateLimitingOffsetRangeInputStream` check evaluates to false.

It really seems like invoking `close()` on any open streams is the right thing to do here. With the other changes in this PR (ref-counting and the `RunOnce`) can we make this work with the previous code that actually invoked close?

Reply:
`close()` on the individual stream is still being invoked. We have just removed it from `RemoteTransferContainer` as in s3 plugin close is already being invoked [here](https://github.com/opensearch-project/OpenSearch/commit/fcf8c25cef4a50649060c85c110c034da9696de0#diff-865f98e22f27e46f34144f074fbe290fbcd93c3ae52d13ddd0d63661123cccc5R187) on individual stream. That is where streams are actually being created. Earlier close wasn't there in S3 and we were doing a bulk close in this class but this approach would delay closing a stream which has already been consumed fully.

Reply:
I have explained [here](https://github.com/opensearch-project/OpenSearch/pull/11334#discussion_r1407032962) on why we need global variable `closed` to be set in `RemoteTransferContainer#close`